lib/rules/array.js

```diff
@@ -175,12 +175,31 @@ class Length extends Rule {
 }
 
 class Unique extends Rule {
+    compile(params) {
+        const comp = params.args[0];
+        if (typeof comp === 'string') {
+            this._key = comp;
+        } else if (comp instanceof Function) {
+            this._compFn = comp;
+        }
+    }
+
     validate(params, callback) {
         const v = params.value;
         const l = v.length;
         for (let i = 0; i < l; i++) {
             for (let k = i + 1; k < l; k++) {
-                if (eq(v[i], v[k])) {
+                let isEqual = false;
+                let compFn = eq;
+                if (this._compFn) {
+                    compFn = this._compFn;
+                }
```

Comment:
Also minor: `const compare = this._compFn || eq` outside the loop. No need to re-run it on every iteration